bundles/ui/org.eclipse.smarthome.ui.icon/src/main/java/org/eclipse/smarthome/ui/icon/internal/IconServlet.java

```diff
@@ -126,6 +126,20 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
                 topProvider = provider;
             }
         }
+        if (topProvider == null) {
+            logger.debug(
+                    "Requested icon category {} provided by no icon provider; try to failback to category \"none\"",
+                    category);
+            // Try to failback to category "none"
+            category = "none";
+            for (IconProvider provider : iconProvider) {
```

Comment:
You are duplicating the logic here - it would probably be much cleaner to extract that into a method, wdyt?